lib/vdsm/storage/mpathconf.py

```diff
@@ -39,9 +44,55 @@
 
 """
 
+# The first line of multipath.conf configured by vdsm must contain a
+# "VDSM REVISION X.Y" tag.  Note that older version used "RHEV REVISION X.Y"
+# format.
+
+CURRENT_TAG = "# VDSM REVISION 2.2"
+
+_OLD_TAGS = (
+    "# VDSM REVISION 2.1",
+    "# VDSM REVISION 2.0",
+    "# VDSM REVISION 1.9",
+    "# VDSM REVISION 1.8",
+    "# VDSM REVISION 1.7",
+    "# VDSM REVISION 1.6",
+    "# VDSM REVISION 1.5",
+    "# VDSM REVISION 1.4",
+    "# VDSM REVISION 1.3",
+    "# VDSM REVISION 1.2",
+    "# RHEV REVISION 1.1",
+    "# RHEV REVISION 1.0",
+    "# RHEV REVISION 0.9",
+    "# RHEV REVISION 0.8",
+    "# RHEV REVISION 0.7",
+    "# RHEV REVISION 0.6",
+    "# RHEV REVISION 0.5",
+    "# RHEV REVISION 0.4",
+    "# RHEV REVISION 0.3",
+    "# RHAT REVISION 0.2",
+)
+
+# The second line of multipath.conf may contain PRIVATE_TAG. This means
+# vdsm-tool should never change the conf file even when using the --force flag.
+
+_PRIVATE_TAG = "# VDSM PRIVATE"
+_OLD_PRIVATE_TAG = "# RHEV PRIVATE"
+
 
 log = logging.getLogger("storage.mpathconf")
 
+MultipathConf = namedtuple("MultipathConf", "revision,private")
+
+
+class Revision(Enum):
+    # Revision number appears in the configuration and is up to date.
+    ok = auto()
+    # Revision number appears in the configuration but is outdated.
+    old = auto()
+    # Revision number does not appear in the configuration.
+    missing = auto()
```

Comment:
These are three variables that are related, so to me, it makes sense to put them in a collection such as an `Enum`. Also, the value of the variables doesn't really matter. With `Enum` we can use the more explicit canonical name instead of arbitrary data, such as strings, and not worry about their value.

I don't see why using magic strings makes the code more clear. Quite the opposite imo, makes people guess if they relate and have to infer it from the code and the variable name.

Reply:
code consistency is a stronger preference than nuances in usage. There are so many of these constants in areas that no one knows by heart and that makes it difficult to replace in one go. Still, that would be the only way how to introduce such a change (if it would be worth it, which it is likely not). Having two different styles throughout the code is the worst option.